File: backend/dotnet/core/RestBioAspNetCoreSample/Controllers/SampleIdentificationController.cs
using Lacuna.RestPki.Api.Bio.Identifications;
using Lacuna.RestPki.Api.Bio.Sessions;
using Lacuna.RestPki.Client;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Options;
using RestBioAspNetCoreSample.Configuration;

namespace RestBioAspNetCoreSample.Controllers {

	[ApiController]
	[Route("/sample-api/sessions")]
	public class SampleIdentificationController(

		IRestBioService restBioService,
		IOptions<ExampleConfig> exampleConfig

	) : ControllerBase {

		[HttpPost("identification2d")]
		public async Task<BioIdentificationResponse> Identification2DAsync(BioIdentificationRequest request, Guid? subscriptionId = null) {
			return await restBioService.IdentifyAsync(request, subscriptionId);
		}

		[HttpPost("identification")]
		public async Task<StartBioSessionResponse> StartIdentificationAsync() {

			// This is an example of how to start an identification session.
			// You must implement your own security measures to ensure that only users
			// you want to have access to this endpoint can call it.

			// The response of the following call contains the session URL that
			// will be loaded in the Widget to start the biometric session.

			var response = await restBioService.StartIdentificationSessionAsync(new StartBioIdentificationSessionRequest {
				TrustedOrigin = exampleConfig.Value.TrustedOrigin,
				// Additional properties for identification session:
				FaceCaptureProvider = Lacuna.RestPki.Api.FaceCaptureProviders.FaceTecLiveness3d
			});


			// Although not mandatory, you may want to save the SessionId in your database
			// along with your user/session information, as you can use this ID to retrieve
			// the status of the session by later using the GetIdentificationSessionCompletionAsync()
			// method. But at the end of the session, the Widget will return you a ticket
			// that you can use to retrieve the session status without needing to store
			// the SessionId.

			// Available properties of the response:
			_ = response.SessionUrl;    // The URL to be loaded in the Widget to start the biometric session.
			_ = response.SessionId;     // The ID of the session.
			_ = response.SessionType;   // The type of the session (IdentificationDocumentCapture).

			return response;
		}

		[HttpPost("identification/completion")]
		public async Task<BioIdentificationSessionStatusModel> GetIdentificationSessionCompletionAsync(CompleteBioSessionRequest request) {

			// This is an example of how to complete an identification session.
			// You must implement your own security measures to ensure that only users
			// you want to have access to this endpoint can call it.

			// By calling the following endpoint, you will get the final status of the
			// biometric session.
			var result = await restBioService.CompleteIdentificationSessionAsync(request.Ticket);

			// Available properties of the result:
			var sessionId = result.SessionId;       // The ID of the session.
			var success = result.Success;           // Whether the biometric session was successful or not.

			if (success == true) {
				// The biometric session was successful and the identification process is complete.
				// You can now trust that the person behind the camera is who they claim to be.

			} else if (success == false) {
				// The biometric session was completed, but the identification failed.
				// Here you may want to retry, or implement other security measures.

			} else {
				// The biometric session is still in progress. This should not happen here,
				// as the Widget will only provide a ticket when the session is finalized.
			}

			return result;
		}

		[HttpGet("identification/status")]
		public async Task<BioIdentificationSessionStatusModel> GetIdentificationSessionStatusAsync(Guid sessionId) {
			return await restBioService.GetIdentificationSessionStatusAsync(sessionId);
		}

	}
}

File: backend/java/src/main/java/com/lacunasoftware/demo/api/controller/SampleIdentificationController.java
package com.lacunasoftware.demo.api.controller;

import java.util.UUID;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.lacunasoftware.demo.config.ExampleConfigProperties;
import com.lacunasoftware.restpkicore.BioIdentificationSessionStatusModel;
import com.lacunasoftware.restpkicore.CompleteBioSessionRequest;
import com.lacunasoftware.restpkicore.FaceCaptureProviders;
import com.lacunasoftware.restpkicore.RestBioService;
import com.lacunasoftware.restpkicore.StartBioIdentificationSessionRequest;
import com.lacunasoftware.restpkicore.StartBioSessionResponse;

@RestController
@RequestMapping("sample-api/sessions/identification")
public class SampleIdentificationController {

    private final RestBioService service;
    private final ExampleConfigProperties exampleConfig;

    public SampleIdentificationController(
        RestBioService restBioService,
        ExampleConfigProperties exampleConfig
    ) {
        this.service = restBioService;
        this.exampleConfig = exampleConfig;
    }

    @PostMapping
    public ResponseEntity<StartBioSessionResponse> StartIdentificationSession() throws Exception {

        // This is an example of how to start an identification session.
        // You must implement your own security measures to ensure that only users
        // you want to have access to this endpoint can call it.

        // The response of the following call contains the session URL that
        // will be loaded in the Widget to start the biometric session.

        // For identification sessions, you don't need to pass a subject identifier,
        // as the goal of this session is to identify the user among all enrolled users.

        var request = new StartBioIdentificationSessionRequest();
        request.setTrustedOrigin(exampleConfig.getTrustedOrigin());
        // Additional properties for identification session:
        request.setFaceCaptureProvider(FaceCaptureProviders.FACETECLIVENESS3D);

        // Although not mandatory, you may want to save the SessionId in your database
        // along with your user/session information, as you can use this ID to retrieve
        // the status of the session by later using the
        // GetIdentificationSessionCompletionAsync()
        // method. But at the end of the session, the Widget will return you a ticket
        // that you can use to retrieve the session status without needing to store
        // the SessionId.

        var response = service.StartIdentificationSession(request);

        // Available properties of the response:
        response.getSessionUrl();   // The URL to be loaded in the Widget to start the biometric session.
        response.getSessionId();    // The ID of the session.
        response.getSessionType();  // The type of the session (Identification).

        return ResponseEntity.ok(response);
    }

    @PostMapping("completion")
    public ResponseEntity<BioIdentificationSessionStatusModel> GetIdentificationSessionCompletion(@RequestBody CompleteBioSessionRequest request) throws Exception {
        // This is an example of how to complete an identification session.
        // You must implement your own security measures to ensure that only users
        // you want to have access to this endpoint can call it.

        // By calling the following endpoint, you will get the final status of the
        // biometric session.
        var result = service.CompleteIdentificationSession(request.getTicket());

        // Available properties of the result:
        result.getSessionId();                      // The ID of the session.
        var success = result.isSuccess();           // Whether the biometric session was successful or not.

        if (Boolean.TRUE.equals(success)) {
            // The biometric session was successful and the identification process is complete.
            // You can now trust that the person behind the camera is who they claim to be.

        } else if (Boolean.FALSE.equals(success)) {
            // The biometric session was completed, but the identification failed.
            // Here you may want to retry, or implement other security measures.

        } else {
            // The biometric session is still in progress. This should not happen here,
            // as the Widget will only provide a ticket when the session is finalized.
        }

        return ResponseEntity.ok(result);
    }

    @GetMapping("status")
    public ResponseEntity<BioIdentificationSessionStatusModel> GetIdentificationSessionStatus(@RequestParam UUID sessionId) throws Exception {
        var status = service.GetIdentificationSessionStatus(sessionId);
        return ResponseEntity.ok(status);
    }

}
